configurator/configurator.go

```diff
@@ -12,15 +12,7 @@ import (
 const confFileName = "cleaner_config.yml"
 
 var (
-	defaultStartPath              = "."
-	defaultBlackList              = ExtList{"lnk", "ini2", "bin", "tmp"}
-	defaultWhiteListDocs          = ExtList{"doc", "docx", "xls", "xlsx", "ppt", "pptx", "pdf"}
-	defaultWhiteListImgs          = ExtList{"png", "png", "jpg", "jpeg", "raw"}
-	defaultWhiteListExts          = slices.Concat(defaultWhiteListDocs, defaultWhiteListImgs)
-	defaultBlackListFiles         = FileList{"~.ini2"}
-	defaultSizeLimit        int64 = 5_000_000
-	defaultContentBlacklist       = []string{"powershell"}
-	Config                  Configuration
+	Config Configuration
```

Comment:
Good start in making variables local! I recommend to apply it to `Config` too. Though it may require wider changes in other files, it will make functions using `Config` more testable.

Reply:
Just to clarify. Now it is set with specific data on app run (on initialization) and we retain values inside this package in memory as we need this value throughout all lifetime. 

So you suggest  to remove this variable from here and...? And set new variable in `main` package? so [setDefaultConf](https://github.com/olegsobchuk/go-cleaner/pull/4/files#diff-66549dfaa5c96d5c7596d5c61d6fdcb8612aaf42a807d3384a0f112552c3f38dR68) should return built `struct`. Correct?

Reply:
I suggest to create a function `func getConfiguration(filePath sting) *Configuration`, that would be called in `main()` function and would return a configuration either read from the existing config file or from default settings set. The configuration pointer then would be passed to other functions if needed. It would allow to override configuration in function tests with mocked configuration to get wanted behaviour during testing.

Reply:
Do you mean change [Iniit](https://github.com/olegsobchuk/go-cleaner/blob/master/configurator/configurator.go#L54) function to 
```
func Init() Configuration {
	if isConfExist() {
		return readConfFile()
	}
		return addConfigFile()
}
```
and use it in `main` function?

Reply:
Well, it may be said this way. If `Init()` function is renamed, external state usage is removed from there, returned value is added, side effect in form of file creation is removed, then it might look like this
```
func getConfiguration(filePath sting) *Configuration {
	if fileExists(filePath) {
		return readConfigFromFile(filePath)
	}
	return getDefaultConfig()
}
```